arangod/Graph/Graph.cpp

```diff
@@ -550,6 +596,33 @@ bool Graph::hasOrphanCollection(std::string const& collectionName) const {
   return orphanCollections().find(collectionName) != orphanCollections().end();
 }
 
+bool Graph::renameCollections(std::string const& oldName, std::string const& newName) {
+  // rename not allowed in a smart collection
+  if (isSmart()) {
+    return false;
+  }
+
+  bool renamed = false;
+
+  // rename collections found in edgeDefinitions
+  for (auto &it : edgeDefinitions()) {
+    if (!renamed) {
+      renamed = it.second.renameCollections(oldName, newName);
+    } else {
+      it.second.renameCollections(oldName, newName);
+    }
+  }
+
+  // orphans
+  if (hasOrphanCollection(oldName)) {
+    _orphanColls.erase(oldName);
+    _orphanColls.insert(oldName);
```

Comment:
this makes no sense... first deleting the item and then inserting the same item again.

Reply:
Thanks. Good you found it. Also there’s a test missing for that scenario. Will fix.

Reply:
:+1: